src/prefect/core/task.py

```diff
@@ -984,6 +983,20 @@ def __iter__(self) -> Iterator:
                 "pass `nout` to the task decorator/constructor, or provide a "
                 "`Tuple` return-type annotation to your task."
             )
+        flow: Flow = prefect.context.get("flow", None)
+        if flow and flow.is_mapped(self):
```

Comment:
If this runs outside of a flow context, then it will trigger the Line 1000 incorrectly; maybe a better way to handle this is to raise a `ValueError` if a flow is not found in context with a message like "Tasks cannot be unpacked outside of a Flow context."

Reply:
The old implementation executes what currently is the line 1000 as well. Further down the call stack, `task.bind` raises an exception that instructs to use an active flow context. That behavior is consistent between both master and this PR.

Reply:
Right, but the catch is that line 1000 is _only_ valid for the old implementation, so it shouldn't be run or relied upon in this new implementation - raising an explicit error earlier on avoids depending on that deeper behavior which may change unexpectedly.